lab-2/website/js/user-controller.js

```diff
@@ -18,40 +18,56 @@ var userController = {
         this.uiElements.profileImage = $('#profilepicture');
 
         this.data.config = config;
-        this.data.auth0Lock = new Auth0Lock(config.auth0.clientId, config.auth0.domain);
+        var params = {
+            autoclose: true,
+            auth: {
+                params: {
+                    scope: 'openid profile email user_metadata picture'
+                },
+                responseType: 'id_token token'
+            }
+        };
+        this.data.auth0Lock = new Auth0Lock(config.auth0.clientId, config.auth0.domain, params);
 
         // check to see if the user has previously logged in
-        var idToken = localStorage.getItem('userToken');
-
-        if (idToken) {
-            this.configureAuthenticatedRequests();
-
-            var that = this;
-
-            this.data.auth0Lock.getProfile(idToken, function (err, profile) {
-                if (err) {
-                    return alert('There was an error getting the profile: ' + err.message);
-                }
-                // Display user information
-                that.showUserAuthenticationDetails(profile);
-
-            });
-        }
+        var accessToken = localStorage.getItem('accessToken');
+        var idToken = localStorage.getItem('idToken');
 
         this.wireEvents();
+        
+        if (accessToken && idToken) {
+            this.getUserProfile(accessToken, idToken);
+        }        
+        
     },
     configureAuthenticatedRequests: function () {
         $.ajaxSetup({
             'beforeSend': function (xhr) {
-                xhr.setRequestHeader('Authorization', 'Bearer ' + localStorage.getItem('userToken'));
+                xhr.setRequestHeader('Authorization', 'Bearer ' + localStorage.getItem('idToken'));
             }
         });
     },
+    getUserProfile: function (accessToken, idToken) {
+        var that = this;
+        this.data.auth0Lock.getUserInfo(accessToken, function(error, profile) {
+            
+            if (error) {
+                return alert('There was an error getting the profile: ' + err.message);
+            }
+            
+            localStorage.setItem('accessToken', accessToken);
```

Comment:
I think these need to be set in this.data.auth0Lock.on("authenticated", function(authResult) and not in a function called getUserProfile